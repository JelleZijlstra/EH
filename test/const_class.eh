#!/usr/bin/ehi

# Test that a class name cannot overwrite a const variable
const Foo = 4

try
	class Foo; end
catch if exception.isA ConstError
	echo exception
end

echo Foo
