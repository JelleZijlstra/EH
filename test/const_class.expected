%%stdout%%
<ConstError: Cannot set constant member Foo in object of type GlobalObject: (global execution context)>
4
%%stderr%%
